sumologic/resource_sumologic_cse_rule_tuning_expression.go

```diff
@@ -0,0 +1,161 @@
+		Read:   resourceSumologicCSERuleTuningExpressionRead,
+		Delete: resourceSumologicCSERuleTuningExpressionDelete,
+		Update: resourceSumologicCSERuleTuningExpressionUpdate,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"description": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: false,
+			},
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: false,
+			},
+			"expression": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: false,
+			},
+			"enabled": {
+				Type:     schema.TypeBool,
+				Required: true,
+				ForceNew: false,
+			},
+			"exclude": {
+				Type:     schema.TypeBool,
+				Required: true,
+				ForceNew: false,
+			},
+			"is_global": {
+				Type:     schema.TypeBool,
+				Required: true,
+				ForceNew: false,
+			},
+			"rule_ids": {
+				Type:     schema.TypeList,
+				Required: true,
+				ForceNew: false,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+		},
+	}
+}
+
+func resourceSumologicCSERuleTuningExpressionRead(d *schema.ResourceData, meta interface{}) error {
+	c := meta.(*Client)
+
+	var CSERuleTuningExpressionGet *CSERuleTuningExpression
+	id := d.Id()
+
+	CSERuleTuningExpressionGet, err := c.GetCSERuleTuningExpression(id)
+	if err != nil {
+		log.Printf("[WARN] CSE Insights Status not found when looking by id: %s, err: %v", id, err)
+
+	}
+
+	if CSERuleTuningExpressionGet == nil {
+		log.Printf("[WARN] CSE Insights Status not found, removing from state: %v - %v", id, err)
+		d.SetId("")
+		return nil
+	}
+
+	d.Set("name", CSERuleTuningExpressionGet.Name)
+	d.Set("description", CSERuleTuningExpressionGet.Description)
+	d.Set("expression", CSERuleTuningExpressionGet.Expression)
+	d.Set("enabled", CSERuleTuningExpressionGet.Enabled)
+	d.Set("exclude", CSERuleTuningExpressionGet.Exclude)
+	d.Set("is_global", CSERuleTuningExpressionGet.IsGlobal)
+	d.Set("rule_Ids", CSERuleTuningExpressionGet.RuleIds)
+
+	return nil
+}
+
+func resourceSumologicCSERuleTuningExpressionDelete(d *schema.ResourceData, meta interface{}) error {
+	c := meta.(*Client)
+
+	return c.DeleteCSERuleTuningExpression(d.Id())
+
+}
+
+func resourceSumologicCSERuleTuningExpressionCreate(d *schema.ResourceData, meta interface{}) error {
+	c := meta.(*Client)
+
+	if d.Id() == "" {
+		id, err := c.CreateCSERuleTuningExpression(CSERuleTuningExpression{
+			Name:        d.Get("name").(string),
+			Description: d.Get("description").(string),
+			Expression:  d.Get("expression").(string),
+			Enabled:     d.Get("enabled").(bool),
+			Exclude:     d.Get("exclude").(bool),
+			IsGlobal:    d.Get("is_global").(bool),
+			RuleIds:     resourceRuleIdsToStringArray(d.Get("rule_ids").([]interface{})),
+		})
+
+		if err != nil {
+			return err
+		}
+		log.Printf("[INFO] got id: %s", id)
+		d.SetId(id)
+	}
+
+	return resourceSumologicCSERuleTuningExpressionUpdate(d, meta)
```

Comment:
can we just do `return resourceSumologicCSERuleTuningExpressionRead(d, meta)`? Why return update here?